cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/executor/InMemoryDatasetOpExecutor.java

```diff
@@ -61,6 +61,9 @@ public DatasetSpecification create(Id.DatasetInstance datasetInstanceId, Dataset
 
     // TODO: Note - for now, just sending the name. However, type likely needs to be namesapce-aware too.
     DatasetSpecification spec = type.configure(datasetInstanceId.getId(), props);
+    if (props.getDescription() != null) {
+      spec = spec.setDescription(props.getDescription());
+    }
```

Comment:
I agree, to be consistent I'll move patching the description to the same place where originalProperties are set.